qos-monitor/src/main/java/eu/arrowhead/core/qos/QoSMonitorApplicationInitListener.java

```diff
@@ -29,24 +33,83 @@
+import eu.arrowhead.core.qos.service.ping.monitor.PingEventCollectorTask;
+import eu.arrowhead.core.qos.service.ping.monitor.PingMonitorManager;
+import eu.arrowhead.core.qos.service.ping.monitor.impl.DefaultExternalPingMonitor;
+import eu.arrowhead.core.qos.service.ping.monitor.impl.DummyPingMonitor;
+import eu.arrowhead.core.qos.service.ping.monitor.impl.OrchestratedExternalPingMonitor;
 
 @Component
 public class QoSMonitorApplicationInitListener extends ApplicationInitListener {
-	
+
 	//=================================================================================================
 	// members
-	
+
 	@Value(CoreCommonConstants.$QOS_IS_GATEKEEPER_PRESENT_WD)
 	private boolean gatekeeperIsPresent;
 
+	@Value(CoreCommonConstants.$QOS_MONITOR_PROVIDER_TYPE_WD)
+	private MonitorProviderType monitorType;
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+	@Bean(QosMonitorConstants.EVENT_QUEUE)
+	public LinkedBlockingQueue<EventDTO> getEventQueue() {
+		logger.debug("getEventQueue started...");
+
+		return new LinkedBlockingQueue<EventDTO>();
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	@Bean(QosMonitorConstants.EVENT_BUFFER)
+	public ConcurrentHashMap<UUID, PingEventBufferElement> getEventBuffer() {
+		logger.debug("getEventBuffer started...");
+
+		return new ConcurrentHashMap<UUID, PingEventBufferElement>();
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	@Bean(QosMonitorConstants.EVENT_COLLECTOR)
+	public PingEventCollectorTask getEventCollector() {
+		logger.debug("getEventCollector started...");
+
+		return new PingEventCollectorTask();
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	@Bean(CoreCommonConstants.PING_MONITOR)
+	public PingMonitorManager getPingMonitor() {
+		logger.debug("getPingMonitor started...");
+
+		switch (monitorType) {
+		case DUMMY:
+			return new DummyPingMonitor();
+		case DEFAULTEXTERNAL:
+			return new DefaultExternalPingMonitor();
+		case ORCHESTRATEDEXTERNAL:
+			return new OrchestratedExternalPingMonitor();
+		default:
+			throw new InvalidParameterException("Not implemented monitor type: " + monitorType.name());
+		}
+
+	}
+
 	//=================================================================================================
 	// assistant methods
 
 	//-------------------------------------------------------------------------------------------------
 	@Override
 	protected List<CoreSystemService> getRequiredCoreSystemServiceUris() {
-		final List<CoreSystemService> result = new ArrayList<>(5);
-		
+		logger.debug("getRequiredCoreSystemServiceUris started...");
+
+		final List<CoreSystemService> result = new ArrayList<>(8);
```

Comment:
8 is not the correct number. Gatekeepr stuff is 3 and based on the monitor type you add 0, 2 or 3 more. That's 6 by using the maximum.

Reply:
fixed